0_RootFS/llvm_common.jl

```diff
@@ -135,7 +135,7 @@ function llvm_script(;version = v"8.0.1", llvm_build_type = "Release", kwargs...
     CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=${LLVM_BUILD_TYPE})
 
     # We want a lot of projects
-    CMAKE_FLAGS+=(-DLLVM_ENABLE_PROJECTS='clang;polly;lld')
+    CMAKE_FLAGS+=(-DLLVM_ENABLE_PROJECTS='clang;flang;lld;mlir;openmp;polly')
```

Comment:
As I abundantly requested in https://github.com/JuliaPackaging/Yggdrasil/pull/10132, please make such changes version conditional.  We always need to be able to rebuild old versions, and requesting flang with llvm 6 doesn't make any sense.

Reply:
Also, why adding mlir and openmp?  We already have openmp as a separate package and I don't think adding runtime would work at all with the current setup of this toolchain

Reply:
MLIR is probably because Flang is built on top of MLIR for some of its internals, and I am guessing we need to make sure the versions match properly to get working tools.

(the best reference I have for the Flang MLIR relation is https://arxiv.org/abs/2409.18824)

Reply:
I have added if statements.

Building flang requires mlir.

I removed openmp.